javascript/javascript2/week2/index.js

```diff
@@ -0,0 +1,130 @@
+// Doubling of number
+
+let numbers = [1, 2, 3, 4];
+
+const newNumbers = numbers
+  .filter(function (params) {
+    return params % 2 !== 0;
+  })
+  .map(function (params) {
+    return params * 2;
+  });
+
+console.log("The doubled numbers are", newNumbers);
+// console.log("The doubled numbers are", newNumbers); // [2, 6]
+
+// Working with movies
+
+//1
+const moviesSort = movies.filter(function (params) {
+  return params.title.length < 5;
+});
+
+console.log(moviesSort);
+
+//2
+const moviesLong = movies.filter(function (params) {
+  return params.title.length > 5;
+});
+
+console.log(moviesLong);
+
+//3
+const moviesBetween = movies.filter(function (params) {
+  return params.year >= 1980 && params.year <= 1989;
+});
+
+console.log(moviesBetween.length);
+
+//4
+const tagMovie = movies.map((element) => {
+  if (element.rating >= 7) {
+    element.tag = "Good";
+  } else if (element.rating >= 4 && element.rating < 7) {
```

Comment:
Since the conditions in if statement are being evaluated from top to bottom, the second condition will get evaluated only if the the rating doest not fit the first one meaning that it is smaller than 7 and therefore you can save some key strokes by not writing the multiple condition in your "else if" 🥷

Reply:
Fixed